### PR TITLE
Feature/consent

### PR DESCRIPTION
**Pull request by oharsta**
Consent feature. See https://www.pivotaltracker.com/n/projects/2326224

### CODE REVIEW

File: src/main/java/oidc/endpoints/AuthorizationEndpoint.java [hunk 1]
```diff
@@ -64,27 +63,26 @@
...
 import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
 
 @Controller
 public class AuthorizationEndpoint implements OidcEndpoint {
 
     private static final Log LOG = LogFactory.getLog(AuthorizationEndpoint.class);
-    private static final List<String> forFreeOpenIDScopes = Arrays.asList("profile", "email", "address", "phone");
+    private static final List<String> forFreeOpenIDScopes = Arrays.asList("openid", "profile", "email", "address", "phone");
```
Comment (quartje) [inline src/main/java/oidc/endpoints/AuthorizationEndpoint.java:86]: Is openid obligatory for client credentials clients now? Or can you just use these at will?

File: src/main/java/oidc/endpoints/JwkKeysEndpoint.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import com.google.crypto.tink.CleartextKeysetHandle;
 import com.google.crypto.tink.JsonKeysetWriter;
 import com.google.crypto.tink.KeysetHandle;
-import com.google.crypto.tink.aead.AeadKeyTemplates;
+import com.google.crypto.tink.aead.AesCtrHmacAeadKeyManager;
```
Comment (quartje) [inline src/main/java/oidc/endpoints/JwkKeysEndpoint.java:7]: Why do we need to change the crypto stuff for the consent feature?

**Review by quartje — COMMENTED** [review]
I've added some comments here and there. Seems like a big change, which will require some thorough testing ..

File: src/main/java/oidc/endpoints/JwkKeysEndpoint.java (hunk 2 above)
Comment (oharsta) [reply src/main/java/oidc/endpoints/JwkKeysEndpoint.java:7]: The methods we used were deprecated. This is the preferred way now. There is very good test coverage for this and it did not break anything

File: src/main/java/oidc/endpoints/AuthorizationEndpoint.java (hunk 1 above)
Comment (oharsta) [reply src/main/java/oidc/endpoints/AuthorizationEndpoint.java:86]: First question: no. Second question: yes. These are the scopes that can be asked for and will be granted regardless of what is configured in manage.

**Pull request merged**